granian/_granian.pyi

```diff
@@ -58,3 +59,35 @@ class WSGIScope:
 
 class WorkerSignal:
     def __init__(self): ...
+
+class __WorkerConfig(Protocol):
+    def __init__(
+        self,
+        id: int,
+        socket_fd: int,
+        threads: int,
+        pthreads: int,
+        http_mode: str,
+        http1_opts: HTTP1Settings,
+        http2_opts: HTTP2Settings,
+        websockets_enabled: bool,
+        opt_enabled: bool,
+        ssl_enabled: bool,
+        ssl_cert: Optional[str],
+        ssl_key: Optional[str],
+    ):
+        ...
+
+class ASGIWorker(__WorkerConfig):
+    ...
+
+class RSGIWorker(__WorkerConfig):
+    ...
+
+class WSGIWorker(__WorkerConfig):
+    ...
+
+
+class ListenerHolder:
+    @classmethod
+    def from_address(cls, bind_addr: str, port: int, backlog: int) -> "ListenerHolder": ...
```

Comment:
I would add `from __future__ import annotations` and remove the string annotation

Reply:
Is __future__ annotations in re of Protocol? 

also unless i'm mistaken, the class method doesn't return the instance?

Reply:
Yup, `ListenerHolder.from_address` returns an instance, see https://github.com/emmett-framework/granian/blob/348d3c71e0e1923688248e2d0744275b7a14129e/src/tcp.rs#L34

I commented about `__future__` so you can rewrite the upper as:

```python
class ListenerHolder:
    @classmethod
    def from_address(cls, bind_addr: str, port: int, backlog: int) -> ListenerHolder: ...
```